src/Compilers/Core/Portable/PEWriter/MetadataWriter.PortablePdb.cs

```diff
@@ -987,5 +987,25 @@ private void EmbedMetadataReferenceInformation(CommonPEModuleBuilder module)
                 kind: _debugMetadataOpt.GetOrAddGuid(PortableCustomDebugInfoKinds.CompilationMetadataReferences),
                 value: _debugMetadataOpt.GetOrAddBlob(builder));
         }
+
+        private void EmbedTypeDocumentInformation(CommonPEModuleBuilder module)
+        {
+            var builder = new BlobBuilder();
+
+            foreach (var (definition, documents) in module.GetTypeToDebugDocumentMap(Context))
```

Comment:
>module.GetTypeToDebugDocumentMap(Context) [](http://example.com/codeflow?start=52&length=41)

I do not think this returns items in any deterministic order. Things should be "sorted" in such a way that order wouldn't change from build to build. This is the likely source of the determinizm failures.